staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -4586,6 +4590,15 @@ func (c *Cloud) nodeNameToProviderID(nodeName types.NodeName) (InstanceID, error
 	return KubernetesInstanceID(node.Spec.ProviderID).MapToAWSInstanceID()
 }
 
+// checkProtocol checks to make sure only TCP is supported for ELB
+func checkProtocol(proto v1.Protocol, annotations map[string]string) error {
+	if !isNLB(annotations) && proto != v1.ProtocolTCP {
```

Comment:
We probably should affirmatively check the supported protocols for NLB; if a new protocol is introduced into k8s that NLB does not support, we probably want to catch it early